docs/synapse/userguide_section3.rst

```diff
@@ -0,0 +1,186 @@
+    ('created',{'ptype':'time:min'}),
+    ('updated',{'ptype':'time:max'}),
+    ('expires',{'ptype':'time:max'}),
+  ]),
+
+**Note** that there are some minor differences between the the auto-generated documentation and the Synapse source code. Since either (or both together) can be helpful for analysts working with Synapse data, it helps to be aware of these differences.
+
+- **Default values.** Some nodes have properties that are automatically set to a specific value unless otherwise specified. If a property has a default value, it will be noted in both docs and the source code.
+
+- **Read-only properties.** Primary properties are unique and cannot be changed. Some secondary properties (typically those derived from the primary property) should also not be modified and are therefore implicitly read-only. In some cases, secondary properties are explicitly defined as read-only in the Synapse source code via the definition ``'ro':1``. However, these designations are not carried over to docs. (An
+ example is the ``:port`` property of an ``inet:url`` node. A port number is generally not included in a URL that uses standard ports for a given protocol (e.g., ``https://www.foo.com/bar/baz.html``). Based on the presence of an “https” prefix in a URL, Synapse will set ``:port=443`` as a read-only property, as specified in the source.)
+
+- **Readability.** While automatic docs are a bit more readable for the general user, the auto-generation process sorts and displays types, forms, and form secondary properties in alphabetical order. However, alphabetical order may not be the most intuitive order for grouping either forms or form-specific properties, based on how an analyst would typically view or work with the data.
+
+  In contrast, the Synapse source code lists forms and form properties in an order that may be more “sensical” for the given node type. The source code also tends to list secondary properties that can be automatically set by Synapse first in the source code (e.g., secondary properties that can be derived from the primary property’s value). For example, when creating the node ``inet:fqdn=woowoo.com``, Synapse can parse that ``<property>=<value>`` and automatically set the secondary properties ``inet:fqdn:domain=com`` and ``inet:fqdn:host=woowoo``. Secondary properties that require that an additional value be provided (e.g., ``inet:fqdn:created``) are listed later in the source code.
+
+**Types**
+
+A **type** is the definition of an element within the data model, describing what the element is and how it should be normalized (if necessary) and structured to conform to the model. Synapse supports standard types (such as integers and strings) as well as extensions of these types. From a user standpoint, types are important primarily as they define the primary and secondary properties of forms.
+
+The data model can be extended to define new types by updating or extending the relevant Synapse source code.
```

Comment:
Types are also nodes which may be added to create/modify types dynamically ( similar to forms/props )